### ISSUE TITLE
Show better error messages when blocked

### ISSUE THREAD

**Issue by sweetaibelle** [knighty/fimfiction-issues#279]
If you try to send a pm to someone that's blocked you, or try to post on their user page and other places you can't because of the block, how about having an error message that tells you that's why it isn't working? 

"Unable to send pm because <x> has blocked you?" for example, or even just giving you the error before you get to the point of trying to send the pm.

Neither "Server Error" or "There was a problem with your request" tell you much at all, unless you actually know that's what it means, and it isn't always immediately obvious that someone's blocked you...

**Comment by ocalhoun6** [comment]
Is that what's happening?

Because I sometimes get errors like that when trying to send a PM from one account, then I'll try from the other account and it goes through just fine. 

The weird thing, though, is this happens with people who would have no reason to block me. Usually happens when I'm trying to reply to people who asked me to do a commission.

**Comment by sweetaibelle** [comment]
I don't know that's the only time you'll get those errors, but those are definitely the two errors that you get when you are blocked and try to pm someone or post on their user page. I've experimented by having somepony block me then unblock me for testing purposes.

I suppose it's possible they blocked you by accident, or it could be that something else went wrong, given how generic they are. That's one reason why I'd like specific errors. You shouldn't have to guess if you're being blocked or if fimfiction is acting up.

Honestly, it should probably say they've blocked you on their user card and page, too.

**Comment by ocalhoun6** [comment]
Maybe the idea is to have more of a ghost block, so the blocked user doesn't know about it?

But to really take that to completion, it would be better to have it look like the PM actually went through, but not display it in the recipient's inbox.

**Comment by sweetaibelle** [comment]
Yeah, it's not exactly stealthy if you don't think the message went through. It just mainly means you'll start complaining about the site not working properly, which I can't imagine being the best thing if you are supporting the site.

Of course, if you thought it went through, then you'd have arguments about messages that were supposedly sent but didn't actually go through. I'd much rather it just tell you straight out that you are being blocked.

Sure, you might try to bypass it if you were being blocked with an alt, but you might try that if you think your accounts acting up, too.
